lib/puppet/type/cisco_bgp_neighbor_af.rb

```diff
@@ -387,6 +389,24 @@ def self.title_patterns
     newvalues(:true, :false, :default)
   end
 
+  newproperty(:route_map_in) do
+    desc 'route-map in state. Valid values are a string defining the name ' \
```

Comment:
The problem is that we still don't do early detection and failure with this change.  This would only fail during the config_set when the cli rejects an integer as an invalid route-map name.  We need to explicitly fail using the 'validate' method in the type to give clear user guidance.

validate do |value|
      fail("Property 'route_map_in' must be a string") unless
        value.is_a? String
    end

    munge do |value|
      value = :default if value == 'default'
      value
    end